app/internal/security/ouath2.py

```diff
@@ -0,0 +1,124 @@
+from datetime import datetime, timedelta
+from typing import Union
+
+from app.config import JWT_ALGORITHM, JWT_KEY
+from passlib.context import CryptContext
+from app.database.database import SessionLocal
+from app.database.models import User
+from fastapi import Depends, HTTPException
+from fastapi.security import OAuth2PasswordBearer
+import jwt
+from starlette.requests import Request
+from starlette.responses import RedirectResponse
+from starlette.status import HTTP_401_UNAUTHORIZED
+from .schema import LoginUser
+
+
+JWT_MIN_EXP = 3
+pwd_context = CryptContext(schemes=["bcrypt"])
+oauth_schema = OAuth2PasswordBearer(tokenUrl="/login")
+
+
+async def get_db_user_by_username(username: str) -> User:
+    '''Checking database for user by username field'''
+    session = SessionLocal()
```

Comment:
Use `Depends(get_db)` instead

Reply:
https://github.com/tiangolo/fastapi/issues/1881
I already tried to do it when i built the function. To the best of my knowledge, you can't use Depends in a function,  but only when you define a route. I think the reason why is that Depends needs fastapi instance in order to retrieve the expected return, and not a Depend object.
Any thoughts?  or can i leave it as it is?
![image](https://user-images.githubusercontent.com/71118404/107018333-f8b96b80-67a8-11eb-8f55-a76006e17160.png)

Reply:
From your like it seems it isn't an issue, it's just we shouldn't use `Depends`. Sababa.
I still don't really like the idea of creating a new session.
Can you pass the database session as a parameter?